stage1/diagnostic/diagnostic.h

```diff
@@ -0,0 +1,148 @@
+// Copyright 2014-2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define _GNU_SOURCE
+#include <errno.h>
+#include <fcntl.h>
+#include <grp.h>
+#include <inttypes.h>
+#include <limits.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "elf.h"
+
+static int exit_err;
+#define exit_if(_cond, _fmt, _args...)				\
+	exit_err++;						\
+	if(_cond) {						\
+		fprintf(stderr, "Error: " _fmt "\n", ##_args);	\
+		exit(exit_err);					\
+	}
+#define pexit_if(_cond, _fmt, _args...)				\
+	exit_if(_cond, _fmt ": %s", ##_args, strerror(errno))
+
+#define MAX_DIAG_DEPTH 10
+#define MIN(_a, _b) (((_a) < (_b)) ? (_a) : (_b))
+
+static inline void map_file(const char *path, int prot, int flags, struct stat *st, void **map)
```

Comment:
I did this to avoid write a "diagnostic library" just for a single function. Including a `static inline` function it is just like copy and paste the code at compilation time.